### ISSUE TITLE
OC 10: OC.currentTheme undefined

### ISSUE THREAD

**Issue by fcturner** [owncloud/core#27568]
### Steps to reproduce
1. Set up OC 10 latest beta (https://download.owncloud.org/community/testing/owncloud-10.0.0beta.zip)
2. Create folder `/apps/theme-mynamename` and give it an `appinfo/info.xml` etc.
3. Activate this 'app' and open Files

### Expected behaviour
Just a beautiful list of files and folders

### Actual behaviour
Only summary of filelist is shown, no files.
Web Inspector learns that `OC.currentTheme.name` cannot be determined, as `OC.currentTheme` is undefined (which was created in #26805):

![image](https://cloud.githubusercontent.com/assets/7526559/24703274/d45ae89c-1a02-11e7-819e-99cab21754d0.png)

### Server configuration
**Operating system**:
Ubuntu 16.04 LTS

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
7

**ownCloud version:** (see ownCloud admin page)
```
installed	true
maintenance	false
needsDbUpgrade	false
version	10.0.0.5
versionstring	10.0.0 beta
edition	Community
productname	mythemename
```
Hence, `productname` works! It's the name I set in  the `getName` function of `/apps/theme-myname/defaults.php`.

**Updated from an older ownCloud or fresh install:**
Fresh

**Where did you install ownCloud from:**
-

**Signing status (ownCloud 9.0 and above):**
Ehm, what?

Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
Results
=======
- theme-mythemename
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- theme-example
	- INVALID_HASH
		- appinfo/info.xml

Raw output
==========
Array
(
	[theme-mythemename] => Array
		(
			[EXCEPTION] => Array
				(
					[class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
					[message] => Signature data not found.
				)

		)

	[theme-example] => Array
		(
			[INVALID_HASH] => Array
				(
					[appinfo/info.xml] => Array
						(
							[expected] => fda8bc18c48df3cc4dacebd1ece3e688907066f22fd1842293a549c7bdeb81b90b4b60fdd040ed531449d5eb4bd45232a9716bab3316c60d13783b5793d62a97
							[current] => 82e35b98fff67d54050fb7673080641466b67e2a7714168e2947b1132e6593a2a6323e172fa0ba5da1480d4fa629825009c28caba67fadfa61d611bc2b89dece
						)

				)

		)

)
```

**List of activated apps:**

Default + this theme.

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Chromium 55.0

**Operating system:**
Windows 7 SP1

#### Web server error log
(none)

#### ownCloud log (data/owncloud.log)
(none)

#### Browser log
(as above)

**Comment by fcturner** [comment]
```
- theme-example
	- INVALID_HASH
		- appinfo/info.xml
```
is due to the fact that I renamed this folder from `example-theme` to `theme-example` and edited the `<id>` element in this file too.

**Comment by PVince81** [comment]
@phisch please have a look

**Comment by phisch** [comment]
@fcturner can you post the contents of your `info.xml`?
I can't see a case where `OC.currentTheme.name` could be undefined.

**Comment by fcturner** [comment]
Sure:

```
<?xml version="1.0"?>
<info>
	<id>theme-mythemename</id>
	<name>mythemename</name>
	<description>Test of mythemename.</description>
	<licence>AGPL</licence>
	<author>Fallon Turner</author>
	<version>1.0.0</version>
	<types>
		<theme/>
	</types>
	<dependencies>
		<owncloud min-version="10.0" max-version="10.0" />
	</dependencies>
	<namespace>Theme</namespace>
</info>
```

Copied and edited from `/apps/example-theme/appinfo/info.xml`.

**Comment by fcturner** [comment]
> I can't see a case where OC.currentTheme.name could be undefined.

Where is this being defined and how? What's the source?

**Comment by fcturner** [comment]
Found it. I replaced `/core/js/js.js` in `/apps/theme-mythemename/core/js/js.js`. It's still the OC9.x version.

I know the downside of replacing js files in a theme, but I want to show small files as `123 B`, not as `<1 kB`. Although it *is* available as variable in the function as `skipSmallSizes` as shown below, I can't find a way to set it as an option in config.php for example.

```js
function humanFileSize(size, skipSmallSizes) {
	var humanList = ['B', 'KB', 'MB', 'GB', 'TB'];
```

Anyway, this can be closed. Apologies for my lack of thinking this more through.
